Source/Scene/Instanced3DModel3DTileContentProvider.js

```diff
@@ -83,123 +81,136 @@ define([
-                batchTableResources : batchTableResources,
-                boundingVolume : that._boundingVolume,
-                pickPrimitive : that._tileset,
-                cull : false,
-                url : undefined,
-                headers : undefined,
-                gltf : undefined,
-                basePath : undefined
+        var view = new DataView(arrayBuffer);
+        byteOffset += sizeOfUint32;  // Skip magic number
+
+        //>>includeStart('debug', pragmas.debug);
+        var version = view.getUint32(byteOffset, true);
+        if (version !== 1) {
+            throw new DeveloperError('Only Instanced 3D Model version 1 is supported. Version ' + version + ' is not.');
+        }
+        //>>includeEnd('debug');
+        byteOffset += sizeOfUint32;
+
+        // Skip byteLength
+        byteOffset += sizeOfUint32;
+
+        var batchTableLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var gltfLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var gltfFormat = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var instancesLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var batchTableResources = new Cesium3DTileBatchTableResources(this, instancesLength);
+        this._batchTableResources = batchTableResources;
+        if (batchTableLength > 0) {
+            var batchTableString = getStringFromTypedArray(uint8Array, byteOffset, batchTableLength);
+            batchTableResources.batchTable = JSON.parse(batchTableString);
+            byteOffset += batchTableLength;
+        }
+
+        var gltfView = new Uint8Array(arrayBuffer, byteOffset, gltfLength);
+        byteOffset += gltfLength;
+
+        var instancesDataSizeInBytes = instancesLength * instanceSizeInBytes;
+        var instancesView = new DataView(arrayBuffer, byteOffset, instancesDataSizeInBytes);
+        byteOffset += instancesDataSizeInBytes;
+
+        // Create model instance collection
+        var collectionOptions = {
+            instances : new Array(instancesLength),
+            batchTableResources : batchTableResources,
+            boundingVolume : this._boundingVolume,
+            tileset : this._tileset,
+            cull : false,
+            url : undefined,
+            headers : undefined,
+            gltf : undefined,
+            basePath : undefined
+        };
+
+        //>>includeStart('debug', pragmas.debug);
+        if((gltfFormat !== 0) && (gltfFormat !== 1)) {
+            throw new DeveloperError('Only glTF format 0 (url) or 1 (embedded) are supported. Format ' + gltfFormat + ' is not');
+        }
+        //>>includeEnd('debug');
+
+        if (gltfFormat === 0) {
+            var gltfUrl = getStringFromTypedArray(gltfView);
+            var url = (new Uri(gltfUrl).isAbsolute()) ? gltfUrl : this._tileset.url + gltfUrl;
```

Comment:
We talked about this earlier, but do we actually want to support data `uri`? That's sort of the point of embedding the glTF in the tile rather than referencing an external resource.

The same question applies to tiles.json which points to tiles by `url`. Should we support `uri` there too?